src/utils/databaseUtil.ts

```diff
@@ -1,36 +1,86 @@
-import { existsSync, readFileSync, writeFileSync, unlinkSync } from 'fs';
+import fs from 'fs';
 
 export const defaultFilePath = 'saves/data.json';
 
-export const databaseUtil = {
-  hasGameFile(): boolean {
-    return existsSync(defaultFilePath);
-  },
+export type LoadResult =
+  | {
+      data: unknown;
+      error: null;
+    }
+  | {
+      data: null;
+      error: unknown;
+    };
+
+export type SaveResult =
+  | {
+      isSaved: true;
+      error: null;
+    }
+  | {
+      isSaved: false;
+      error: unknown;
+    };
+
+export type DeleteResult =
+  | {
+      isFileOnDisk: false;
+      error: null;
+    }
+  | {
+      isFileOnDisk: boolean;
+      error: unknown;
+    };
 
-  load(): unknown {
+export const databaseUtil = {
+  delete(): DeleteResult {
     try {
-      const data = readFileSync(defaultFilePath, 'utf-8');
-      return JSON.parse(data);
+      fs.unlinkSync(defaultFilePath);
+
+      return {
+        isFileOnDisk: false,
```

Comment:
Should `false` be `databaseUtil.hasGameFile()` here to be extra safe?

Reply:
Nope. I trust `fs.unlinkSync`. 

As a side note I spent way too long trying to write an integration test where the system would reject fs.unlinkSync and I could not (at least on Windows). I just made [this ticket](https://github.com/normal-devs/klicker-knight/projects/1#card-77267526) if we want to investigate it further